apps/ui/components/reservations/UnpaidReservationNotification.tsx

```diff
@@ -57,20 +77,42 @@ function ReservationNotification({
   disabled?: boolean;
   isLoading?: boolean;
 }) {
+  const startRemainingMinutes = reservation.order?.expiresInMinutes;
+  const [remainingMinutes, setRemainingMinutes] = useState(
+    startRemainingMinutes
+  );
   const { t } = useTranslation(["notification, common"]);
+  const isCreated = reservation.state === State.Created;
 
-  const title =
-    reservation.state === State.Created
-      ? t("notification:createdReservation.title")
-      : t("notification:waitingForPayment.title");
-  const text =
-    reservation.state === State.Created
-      ? t("notification:createdReservation.body")
-      : t("notification:waitingForPayment.body");
-  const submitButtonText =
-    reservation.state === State.Created
-      ? t("notification:createdReservation.continueReservation")
-      : t("notification:waitingForPayment.payReservation");
+  const translateKey = isCreated
+    ? "notification:createdReservation"
+    : "notification:waitingForPayment";
+  const title = t(`${translateKey}.title`);
+  const submitButtonText = t(
+    `${translateKey}${isCreated ? ".continueReservation" : ".payReservation"}`
+  );
+  const text = t(`${translateKey}.body`, {
+    time: remainingMinutes,
```

Comment:
just to confirm, this doesn't affect the time of `Created`? since the value would be `0` or `null` here.

Reply:
No it shouldn't. The `Created` variant shows 20 minutes as it's a different translation key. (`waitingForPayment` vs `createdReservation`)